scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -447,11 +460,27 @@ class FossId internal constructor(
             }
 
             checkScan(scanCode, *deltaScanRunParameters(existingScancode))
+
+            enforceDeltaScanLimit(recentScans)
         }
 
         return scanCode
     }
 
+    /**
+     * Make sure that only the configured number of delta scans exists for the current package. Based on the list of
+     * [existingScans], delete older scans until the maximum number of delta scans is reached.
+     */
+    private suspend fun enforceDeltaScanLimit(existingScans: List<Scan>) {
+        log.info { "Number of delta scans to keep: ${config.deltaScanLimit}." }
+        if (existingScans.size >= config.deltaScanLimit) {
```

Comment:
Why is the comparison `>=` here? If we have 10 existing scans, and we're fine to keep 10 scans, then no scan should be deleted. Or is this because the current scan is not yet included in the number of existing scans?

Reply:
Exactly, the current scan is not included in the list; it needs to be counted separately.

Reply:
Thanks for confirming. Can we probably get a code comment for this fact? And maybe change the comparison slightly to `existingScans.size + 1 > config.deltaScanLimit` to emphasize?

Reply:
Added a comment and changed the condition.